src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1113,9 +1114,15 @@ private static Vector128<float> Permute(Vector128<float> value, byte control)
             {
                 return Avx.Permute(value, control);
             }
-
-            Debug.Assert(Sse.IsSupported);
-            return Sse.Shuffle(value, value, control);
+            else if (Sse.IsSupported)
+            {
+                return Sse.Shuffle(value, value, control);
+            }
+            else
+            {
+                // Redundant test so we won't prejit remainder of this method on platforms without AdvSimd.
+                throw new PlatformNotSupportedException();
```

Comment:
We are not optimizing `Permute()` with ARM64 intrinsics? Same for some of the other methods below like `Invert()`,

Reply:
Permute is only used from the x86 code paths. ARM doesn't have a `Permute`/`Shuffle` instruction and the corresponding ARM64 implementation for functions like `Invert` differs quite significantly.